addons/google_calendar/models/res_users.py

```diff
@@ -80,6 +80,14 @@ def _refresh_google_calendar_token(self):
 
     def _sync_google_calendar(self, calendar_service: GoogleCalendarService):
         self.ensure_one()
+
+        # don't attempt to sync when another sync is already in progress, as we wouldn't be
+        # able to commit the transaction anyway (row is locked)
+        self.env.cr.execute("""SELECT id FROM res_users WHERE id = %s FOR UPDATE SKIP LOCKED""", [self.id])
```

Comment:
@odony It locked the update of the foreign, but we only need to write on the rtoken and sync token
what do you think of 
```suggestion
        self.env.cr.execute("""SELECT id FROM res_users WHERE id = %s FOR NO KEY UPDATE SKIP LOCKED""", [self.id])
```

Reply:
yup, even better :+1: